### DISCUSSION TITLE
How to replace existing s3 object as same path

### DISCUSSION THREAD

**Discussion by Zaky7** [aws/aws-sdk-java-v2#3207] [Q&A]
Hi I am getting below error while setting a file to same path again in AWS S3. 

`This copy request is illegal because it is trying to copy an object to itself without changing the object's metadata` 

Although i am update metadata directive
```kotlin
mutableMapOf(
                "x-amz-metadata-directive" to "REPLACE",
         )
```

https://stackoverflow.com/questions/72335494/how-copy-an-existing-object-to-same-path-amazon-s3

Hi i have raised issue on stack over flow also

**Comment by debora-ito** [comment]
The message is a result of a validation on the S3 service side, so maybe the copy request is missing the new metadata headers.

The questions on StackOverflow are relevant though, why do you need to copy the object to the same path?
